File: 백준/Silver/5567. 결혼식/결혼식.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int M = Integer.parseInt(br.readLine());
		boolean[][] rel = new boolean[N + 1][N + 1];

		for(int i = 0; i < M; i++){
			StringTokenizer st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			rel[a][b] = true;
			rel[b][a] = true;
		}

		int cnt = bfs(rel, N);
		System.out.println(cnt);
	}

	private static int bfs(boolean[][] rel, int N) {
		boolean[] visited = new boolean[N + 1];
		Queue<Integer> q = new LinkedList<>();
		q.offer(1);
		visited[1] = true;

		int count = 0;
		int depth = 0;
		while (!q.isEmpty() && depth < 2) {
      int size = q.size();
      for (int i = 0; i < size; i++) {
        int current = q.poll();
        for (int j = 1; j <= N; j++) {
          if (rel[current][j] && !visited[j]) {
            visited[j] = true;
            q.add(j);
            count++;
          }
        }
      }
    depth++;
    }   
    return count;
	}
}

File: 백준/Silver/5567. 결혼식/결혼식.py
from collections import deque

n = int(input())
m = int(input())
rel = [[] for _ in range(n + 1)]

for _ in range(m):
	a, b = map(int, input().split())
	rel[a].append(b)
	rel[b].append(a)

visited = [False] * (n + 1)

def BFS(start):
	q = deque([(start, 0)])
	visited[start] = True
	cnt = 0

	while q:
		cur, depth = q.popleft()

		if depth >= 2:
			continue
		
		for next in rel[cur]:
			if not visited[next]:
				visited[next] = True
				cnt += 1
				q.append([next, depth + 1])
	
	return cnt

print(BFS(1))